gnovm/stdlibs/std/coins.gno

```diff
@@ -13,28 +14,147 @@ type Coin struct {
 	Amount int64  `json:"amount"`
 }
 
+// NewCoin returns a new coin with a denomination and amount
+func NewCoin(denom string, amount int64) Coin {
+	return Coin{
+		Denom:  denom,
+		Amount: amount,
+	}
+}
+
+// String provides a human-readable representation of a coin
 func (c Coin) String() string {
+	if c.IsZero() {
+		return "0" + c.Denom
+	}
```

Comment:
How is this different than always returning `strconv.Itoa(int(c.Amount)) + c.Denom`?